src/snakemake/workflow.py

```diff
@@ -1265,7 +1265,7 @@ def execute(
                 and self.remote_execution_settings.job_deploy_sources
                 and not executor_plugin.common_settings.can_transfer_local_files
             )
-            if should_deploy_sources:
+            if should_deploy_sources and not self.dryrun and len(self.dag):
```

Comment:
as just discussed, the two criteria should be part of the boolean above already.